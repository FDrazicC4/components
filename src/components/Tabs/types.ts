import React, {Node, ReactNode} from "react";

export type Variants = 'standard' | 'icon';
export type Colors = 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info' | 'light' | 'dark' | 'link';
export type Sizes = 'small' | 'medium' | 'large';


export interface Props {
    /**
     * The content inside the tabs to be rendered or the icon to render if `variant="icon"`. Usually just a string, but can be any valid JSX.
     */
    children?: ReactNode,
    /**
     * Classes to be passed through to the component.
     */
    className?: string,
}
