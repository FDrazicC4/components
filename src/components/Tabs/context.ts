import React from 'react';

export const TabContext = React.createContext({});

export const useTabs = (): { value?: String, onChange?: (event?: React.ChangeEvent<{}>, value?: any) => void, shouldPreserveContent?: boolean, } => React.useContext(TabContext);