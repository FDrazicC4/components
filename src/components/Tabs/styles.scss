@import "../../styles/utilities";


// ==============================================
// General
// ==============================================

.wrapper {
    display: block;
    border-bottom: 1px solid #000;
}