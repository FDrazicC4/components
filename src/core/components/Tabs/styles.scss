@import '../../styles/utilities';

.tabs {
	display: flex;
}
