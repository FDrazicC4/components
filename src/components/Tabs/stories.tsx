import React from 'react';
import Tabs from './index';
import TabList from '../TabList';
import Tab from '../Tab';
import TabPanel from '../TabPanel';
import Typography from '../Typography';

export default {
    title: 'In Development|Navigation/Tabs',
};

export const withSimpleTabs = () => {
    const [currentTab, setCurrentTab] = React.useState(0);

    const handleSetTab = (e, i) => {
        setCurrentTab(i);
    };

    return (
        <Tabs value={currentTab} onChange={handleSetTab}>
            <TabList>
                <Tab label="Item One" />
                <Tab label="Item Two" />
                <Tab label="Item Three" />
                <Tab label="Input content" />
            </TabList>
            <TabPanel value={0} style={{ padding: '1rem' }}>
                <Typography variant="body1">Item One Content</Typography>
            </TabPanel>
            <TabPanel value={1} style={{ padding: '1rem' }}>
                <Typography variant="body1">Item Two Content</Typography>
            </TabPanel>
            <TabPanel value={2} style={{ padding: '1rem' }}>
                <Typography variant="body1">Item Three Content</Typography>
            </TabPanel>
            <TabPanel value={3} style={{ padding: '1rem' }}>
                <Typography>Warning! I don't preserve content</Typography>
                <Typography>If you change this input's value and then come back, the value will be lost</Typography>
                <input type="text" placeholder="Change me" />
            </TabPanel>
        </Tabs>
    );
};

export const withIcons = () => {
    const [currentTab, setCurrentTab] = React.useState(0);

    return (
        <Tabs value={currentTab} onChange={(e, i) => {
            setCurrentTab(i);
        }}>
            <TabList>
                <Tab icon="speed" label="Dashboard" />
                <Tab icon="devices" label="Devices"/>
                <Tab icon="settings" label="Site Settings"/>
                <Tab icon="account_circle" label="Client Services"/>
                <Tab icon="description" label="Notes"/>
            </TabList>
            <TabPanel value={0} style={{ padding: '1rem' }}>
                <Typography variant="body1">Dashboard</Typography>
            </TabPanel>
            <TabPanel value={1} style={{ padding: '1rem' }}>
                <Typography variant="body1">Devices</Typography>
            </TabPanel>
            <TabPanel value={2} style={{ padding: '1rem' }}>
                <Typography variant="body1">Site Settings</Typography>
            </TabPanel>
            <TabPanel value={3} style={{ padding: '1rem' }}>
                <Typography variant="body1">Client Services</Typography>
            </TabPanel>
            <TabPanel value={4} style={{ padding: '1rem' }}>
                <Typography variant="body1">Notes</Typography>
            </TabPanel>
        </Tabs>
    )
};

export const withPreservedContent = () => {
    const [currentTab, setCurrentTab] = React.useState(0);

    return (
        <Tabs 
            shouldPreserveContent 
            value={currentTab} 
            onChange={(e, i) => {
                setCurrentTab(i);
            }}
        >
            <TabList>
                <Tab label="Input 1" />
                <Tab label="Input 2"/>
                <Tab label="Input 3"/>
            </TabList>
            <TabPanel value={0} style={{ padding: '1rem' }}>
                <input type="text" placeholder="Change me" />
            </TabPanel>
            <TabPanel value={1} style={{ padding: '1rem' }}>
                <input type="text" placeholder="Change me" />
            </TabPanel>
            <TabPanel value={2} style={{ padding: '1rem' }}>
                <input type="text" placeholder="Change me" />
            </TabPanel>
        </Tabs>
    );
}