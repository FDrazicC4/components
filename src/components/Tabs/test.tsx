import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from './index';


describe('Tabs', () => {

    it('Renders correctly with defaults', () => {
        const tree = renderer
            .create(<Tabs />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
