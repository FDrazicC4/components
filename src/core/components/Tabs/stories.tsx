import React   from 'react';
import { number } from '@storybook/addon-knobs';
import Label from "../Label";
import Tabs, { Tab } from './index';


export default {
    title: 'Components|Display/Tabs',
    component: Tabs,
    parameters: {
        componentSubtitle: <Label color="warning">In Progress</Label>,
    },
};


export const General = () => {
    const tabs = [...Array(3)];
    const activeIndex = number('Active Index', 0, { range: true, min: 0, max: tabs.length - 1, step: 1 });
    return (
        <Tabs>
            {tabs.map((tab, i) => (
                <Tab isActive={activeIndex === i}>Item {(i + 1).toString()}</Tab>
            ))}
        </Tabs>
    );
};


// export const Default = () => <Button>Default</Button>;
// export const Standard = () => <Button variant="standard">Click me!</Button>;
// export const Icon = () => <Button variant="icon">360</Button>;
// export const Circular = () => <Button variant="icon" isCircular>add</Button>;
// export const Loading = () => <Button variant="standard" isLoading>Click me!</Button>;
// export const Fluid = () => <Button variant="standard" isFluid>Click me!</Button>;
// export const Compact = () => <Button variant="standard" isCompact>Click me!</Button>;
// export const Disabled = () => <Button variant="standard" disabled>Click me!</Button>;
