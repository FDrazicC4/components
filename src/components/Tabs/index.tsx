import React from 'react';
import { Switch } from 'react-router-dom';
import clsx from 'clsx';
import { Props } from './types';
import styles from './styles.scss';


/**
 * The `Tabs` component represents a clickable button, which can be used in forms or anywhere in the application
 * that needs simple, standard button functionality.
 */
function Tabs({
    children,
    className,
    ...otherProps
}: Props) {


    const wrapperStyles = clsx(
        [styles.wrapper],
        className,
    );


    return (
        <Switch>
            <div className={wrapperStyles} {...otherProps}>
                {children}
            </div>
        </Switch>
    );


}


export default Tabs;