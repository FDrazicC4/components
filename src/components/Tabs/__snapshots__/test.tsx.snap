// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Renders correctly with defaults 1`] = `
<div
  className="brew-Tabs"
/>
`;
